enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+---
+title: single-node-developer-cluster-profile
+authors:
+  - "@rkukura"
+reviewers:
+  - "@deads2k"
+  - "@LalatenduMohanty"
+  - "@wking"
+approvers:
+  - "@derekwaynecarr"
+  - "@smarterclayton"
+creation-date: 2020-09-18
+last-updated: 2020-10-05
+status: implementable
+see-also:
+replaces:
+superseded-by:
+---
+
+# Single Node Developer Cluster Profile
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Add a new single-node-developer [cluster
+profile](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+that defines the set of OpenShift components, and the configuration
+thereof, applicable to single-node OpenShift Container Platform
+clusters intended for application development as well as for learning
+and exploring OpenShift, but never for production deployment of
+applications. This cluster profile will be utilized in producing the
+Code Ready Containers (CRC) product that runs in a single VM on a
+developer’s workstation or laptop.
+
+In a cluster deployed using this profile, a single node serves as both
+the cluster’s control plane and as a worker node. The cluster profile
+will support the OpenShift components needed for application
+development, including the operator hub and console. In order to
+minimize system resource requirements, the cluster profile will
+exclude production-oriented features such as high availability of the
+control plane components, machine upgradability, and advanced
+telemetry and logging.
+
+This cluster profile will result in a new topology of OpenShift
+components to be officially supported for its intended usage in the
+CRC product. Once the profile is introduced, all teams developing
+OpenShift components will need to be aware of how their components are
+configured in this profile. For many components, this will be simply a
+matter of whether the component is included or excluded. For others,
+specialized configurations and/or features may be used in this
+profile, possibly requiring development and ongoing maintenance
+efforts.
+
+## Motivation
+
+The CRC product’s VM image is currently built using [Single Node
+Cluster (SNC)](https://github.com/code-ready/snc) tooling. The snc.sh
+shell script runs the OpenShift installer to deploy a running cluster
+into a libvirt/KVM environment. One VM serves as the bootstrap node,
+while another serves as the combined master/worker node. Between
+installation steps, and after the installer completes, snc.sh makes
+various modifications to the cluster, such as configuring the etcd
+operator to run without HA, tuning CPU, storage, and networking
```

Comment:
this etcd configuration is currently "highly" unsupported. the etcd team (@hexfusion @ironcladlou) need to aware and add option that if turned on in production clusters can have catastrophic side effects.

Reply:
Having each component explicitly add an annotation to indicate that the component will be used in a given profile is precisely a way of ensuring that the etcd team is aware of what is being done in the single-node profile.

Reply:
I think it would be useful to show an example of such annotation in the context of a component.

Reply:
Agreed that single node non-HA etcd is completely unsupported for production.

During phase 1, we'll just be adding the profile's annotation to existing manifests, with no other changes. 

I'd guess the etcd operator would be one of the first candidates to have a specialized manifest added during phase 2, so I'll add that as an example.

Reply:
On that etcd matter, see https://github.com/openshift/enhancements/pull/504 which has a need for single node clusters in production, and is having the same issue with etcd.